spec.md

```diff
@@ -913,10 +918,16 @@ reg myreg: SInt, myclock
 ; ...
 ```
 
-Optionally, for the purposes of circuit initialization, a register can be
-declared with a reset signal and value. In the following example,
-`myreg`{.firrtl} is assigned the value `myinit`{.firrtl} when the signal
-`myreset`{.firrtl} is high.
+A register may be declared with a reset signal and value. The register's value 
+is updated with the reset value when the reset is asserted.  The reset signal 
+must be a `Reset`{.firrtl}, `UInt<1>`{.firrtl}, or `AsyncReset`{.firrtl}, and 
+the type of initialization  value must be equivalent to the declared type of the 
+register (see [@sec:type-equivalence] for details). The behavior of the register 
+depends on the type of the reset signal.  `AsyncReset`.{firrtl} will immediately 
+change the value of the register, while other types will not change the value of 
```

Comment:
This is only true for synchronous reset. Abstract reset may be either a sync or sync. You can't reason about it until reset inference runs.